enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+commands. This is preferred so that the database design doesn't require a separate database (not server) per cluster for
+least privilege authorization. Note that this HTTP endpoint will only start if the `governance-policy-database` secret
+exists. Technically speaking, a separate goroutine would watch for the creation of this secret.
+
+The authorization code would leverage the Go APIs in
+[github.com/stolostron/rbac-api-utils](https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187)
+or [kube-rbac-proxy](https://pkg.go.dev/github.com/brancz/kube-rbac-proxy@v0.14.2/pkg/authz#NewSarAuthorizer), and
+require the Kubernetes service account recording the compliance event to have write access to the `status` subresource
+of the `policies.policy.open-cluster-management.io` resource in the applicable managed cluster namespace on the hub. The
+Status Sync controller already has this access. Below is an example of a role on the hub that could record compliance
+events for a managed cluster called `cluster1`.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: policy-status-writer
+  namespace: cluster1
+rules:
+  - apiGroups:
+      - policy.open-cluster-management.io
+    resources:
+      - policies/status
+    verbs:
+      - patch
+```
+
+The HTTP endpoint would be at `/api/v1/compliance-events` and would accept a `POST` request with the following schema:
+
+```json
+{
+  "cluster": { "name": "cluster1" },
+  "parent_policy": { "name": "policies.etcd-encryption" },
+  "policy": {
+    "apiGroup": "policy.open-cluster-management.io",
+    "kind": "ConfigurationPolicy",
+    // Optional if specHash is provided and an entry already exists for that spec hash in the database.
+    // This approach saves on network bandwidth since the spec can be quite large.
+    "spec": {},
+    // Optional if spec is provided
+    "specHash": {}
+  },
+  "event": {
+    "compliance": "NonCompliant",
+    "message": "configmaps [app-data] not found in namespace default",
+    "timestamp": "2023-07-19T18:25:43.511Z",
+    // Optional
+    "metadata": {}
+  }
+}
+```
+
+If the compliance event is recorded correctly, an HTTP 201 status code is returned.
+
+##### Viewing Compliance Events
+
+Another HTTP endpoint would be added at `/api/v1/compliance-events` that would accept a `GET` request. This would be
+responsible for exposing the compliance events and performing the correct authorization. The authorization would require
+`get` permissions on the cluster's associated `ManagedCluster` object to be able to list the compliance events for the
+cluster. For performance reasons, this is preferred to performing a `SubjectAccessReview` on every cluster namespace for
+a user per request. We can utilize the
+[github.com/stolostron/rbac-api-utils](https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187)
+library to do this.
```

Comment:
I'm just thinking about this permission. Should everyone that can `get` that ManagedCluster be able to see all the compliance events on all the policies on that cluster?

I don't see a great alternative (it seems like custom rbac verbs aren't a thing, otherwise `get-compliance-events` on the ManagedCluster would be ideal), but maybe a non-resource-url `/compliance-events` would help? Users would need a clusterrole for that (since non-resource-urls are cluster-scoped, I think), and additionally they would need `get` on the `ManagedClusters` they're asking for.